azurerm/resource_arm_application_insights_webtests_test.go

```diff
@@ -0,0 +1,275 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+)
+
+func TestAccAzureRMApplicationInsightsWebTests_basic(t *testing.T) {
+	resourceName := "azurerm_application_insights_webtest.test"
+	ri := tf.AccRandTimeInt()
+	config := testAccAzureRMApplicationInsightsWebTests_basic(ri, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApplicationInsightsWebTestsDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationInsightsWebTestExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "provisioning_state", "Succeeded"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMApplicationInsightsWebTests_complete(t *testing.T) {
+	resourceName := "azurerm_application_insights_webtest.test"
+	ri := tf.AccRandTimeInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApplicationInsightsWebTestsDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApplicationInsightsWebTests_complete(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationInsightsWebTestExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "provisioning_state", "Succeeded"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMApplicationInsightsWebTests_update(t *testing.T) {
+	resourceName := "azurerm_application_insights_webtest.test"
+	ri := tf.AccRandTimeInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApplicationInsightsWebTestsDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApplicationInsightsWebTests_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationInsightsWebTestExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "provisioning_state", "Succeeded"),
+					resource.TestCheckResourceAttr(resourceName, "geo_locations.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "frequency", "300"),
+					resource.TestCheckResourceAttr(resourceName, "timeout", "30"),
+				),
+			},
+			{
+				Config: testAccAzureRMApplicationInsightsWebTests_complete(ri, testLocation()),
```

Comment:
can we pull the value of `testLocation()` out into a variable? it's likely that'll return a different location on each invocation in future